modules/base/translation/keplertranslation.cpp

```diff
@@ -0,0 +1,378 @@
+                "YYYY MM DD HH:mm:ss",
+                Optional::No
+            },
+            {
+                KeyPeriod,
+                new DoubleGreaterVerifier(0.0),
+                "Specifies the orbital period (in seconds).",
+                Optional::No
+            },
+        },
+        Exhaustive::Yes
+    };
+}
+
+KeplerTranslation::KeplerTranslation()
+    : Translation()
+    , _eccentricity("eccentricity", "Eccentricity", 0.0, 0.0, 1.0)
+    , _semiMajorAxis("semimajorAxis", "Semi-major axis", 0.0, 0.0, 1e6)
+    , _inclination("inclination", "Inclination", 0.0, 0.0, 360.0)
+    , _ascendingNode(
+        "ascendingNode",
+        "Right ascension of ascending Node",
+        0.0,
+        0.0,
+        360.0
+    )
+    , _argumentOfPeriapsis(
+        "argumentOfPeriapsis",
+        "Argument of Periapsis",
+        0.0,
+        0.0,
+        360.0
+    )
+    , _meanAnomalyAtEpoch("meanAnomalyAtEpoch", "Mean anomaly at epoch", 0.0, 0.0, 360.0)
+    , _epoch("epoch", "Epoch", 0.0, 0.0, 1e9)
+    , _period("period", "Orbit period", 0.0, 0.0, 1e6)
+    , _orbitPlaneDirty(true)
+{
+    auto update = [this]() {
+        _orbitPlaneDirty = true;
+    };
+
+    // Only the eccentricity, semimajor axis, inclination, and location of ascending node
+    // invalidate the shape of the orbit. The other parameters only determine the location
+    // the spacecraft on that orbit
+    _eccentricity.onChange(update);
+    addProperty(_eccentricity);
+
+    _semiMajorAxis.onChange(update);
+    addProperty(_semiMajorAxis);
+
+    _inclination.onChange(update);
+    addProperty(_inclination);
+
+    _ascendingNode.onChange(update);
+    addProperty(_ascendingNode);
+
+    _argumentOfPeriapsis.onChange(update);
+    addProperty(_argumentOfPeriapsis);
+
+    addProperty(_meanAnomalyAtEpoch);
+    addProperty(_epoch);
+    addProperty(_period);
+}
+
+KeplerTranslation::KeplerTranslation(const ghoul::Dictionary& dictionary) 
+    : KeplerTranslation()
+{
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "KeplerTranslation"
+    );
+
+    setKeplerElements(
+        dictionary.value<double>(KeyEccentricity),
+        dictionary.value<double>(KeySemiMajorAxis),
+        dictionary.value<double>(KeyInclination),
+        dictionary.value<double>(KeyAscendingNode),
+        dictionary.value<double>(KeyArgumentOfPeriapsis),
+        dictionary.value<double>(KeyMeanAnomaly),
+        dictionary.value<double>(KeyPeriod),
+        dictionary.value<std::string>(KeyEpoch)
+    );
+}
+
+glm::dvec3 KeplerTranslation::position() const {
+    return _position;
+}
+
+double KeplerTranslation::eccentricAnomaly(double meanAnomaly) const {
+    // Compute the eccentric anomaly (the location of the spacecraft taking the
+    // eccentricity of the orbit into account) using different solves for the regimes in
+    // which they are most efficient
```

Comment:
Suggest adding more to this comment to explain the different solver criteria for different values of eccentricity.